@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -588,48 +583,35 @@ export class VisibleTileSet {
 
         this.m_dataSourceCache.shrinkToCapacity();
 
-        let minElevation = EarthConstants.MAX_ELEVATION;
-        let maxElevation = EarthConstants.MIN_ELEVATION;
+        let minElevation: number | undefined;
+        let maxElevation: number | undefined;
         this.dataSourceTileList.forEach(renderListEntry => {
             // Calculate min/max elevation from every data source tiles,
             // data sources without elevationRangeSource will contribute to
             // values with zero levels for both elevations.
             const tiles = renderListEntry.visibleTiles;
             tiles.forEach(tile => {
-                minElevation = Math.min(minElevation, tile.minElevation);
-                maxElevation = Math.max(maxElevation, tile.maxElevation);
+                minElevation = MathUtils.min2(minElevation, tile.minElevation);
+                maxElevation = MathUtils.max2(
+                    maxElevation,
+                    tile.maxElevation + tile.maxGeometryHeight
+                );
             });
         });
-        // We process buildings elevation yet (via ElevationRangeSource or either way), so correct
-        // the highest elevation to account for highest possible building.
-        const projectionScale = this.m_frustumIntersection.projection.getScaleFactor(
-            this.m_frustumIntersection.camera.position
-        );
-        const maxBuildingHeight = EarthConstants.MAX_BUILDING_HEIGHT * projectionScale;
-        maxElevation =
-            maxElevation < EarthConstants.MAX_ELEVATION - maxBuildingHeight
-                ? maxElevation + maxBuildingHeight
-                : maxElevation;
-        // Update elevation ranges stored.
-        if (
-            minElevation !== this.m_elevationRange.minElevation ||
-            maxElevation !== this.m_elevationRange.maxElevation
-        ) {
-            this.m_elevationRange.minElevation = minElevation;
-            this.m_elevationRange.maxElevation = maxElevation;
-        }
-        this.m_elevationRange.calculationStatus = visibleTileKeysResult.allBoundingBoxesFinal
-            ? CalculationStatus.FinalPrecise
-            : CalculationStatus.PendingApproximate;
 
+        if (minElevation === undefined) {
+            minElevation = EarthConstants.MIN_ELEVATION;
```

Comment:
Probably:
minElevation = 0;
maxElevation = 0;
We do not wan't to extend planes (instead of required margins) if no elevation info is provided.

Reply:
done